offworld_gym/envs/gazebo/docker_gazebo_env.py

```diff
@@ -0,0 +1,264 @@
+        container_volumes_str = f" -v {XSERVER_VOLUME}:{XSERVER_VOLUME}"
+        container_ports_str = f" -p {ROS_BRIDGE_PORT}:{ROS_BRIDGE_PORT}" \
+                              f" -p {GAZEBO_WEB_SERVER_INTERNAL_PORT}:{GAZEBO_WEB_SERVER_INTERNAL_PORT}" \
+                              f" -p {HTTP_COMMAND_SERVER_PORT}:{HTTP_COMMAND_SERVER_PORT}"
+        docker_run_command = f"docker run --name \'{container_name}\' -it -d --rm " \
+                            f"{container_env_str}{container_volumes_str}{container_ports_str} " \
+                            f"offworldai/offworld-gym:latest {container_entrypoint}"
+        start_container = subprocess.Popen(["/bin/bash", "-c", docker_run_command])
+        logger.debug(f"Docker run command is:\n{docker_run_command}\n")
+        time.sleep(10.0)
+        # get the container id for corresponding container
+        get_container_id_command = f'docker ps -aqf "name={container_name}"'
+        container_id = subprocess.check_output(["/bin/bash", "-c", get_container_id_command]).decode("utf-8").strip()
+        logger.debug(f"Docker container id is:\n{container_id}\n")
+        # get the ip address for corresponding container
+        filter_string = "'{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}'"
+        get_container_ip_command = f"docker inspect -f {filter_string} {container_name}"
+        self._container_ip = subprocess.check_output(["/bin/bash", "-c", get_container_ip_command]).decode("utf-8").strip()
+        logger.debug(f"Docker container ip is:\n{self._container_ip}\n")
+        
+        # ensure cleanup at exit
+        def kill_container_if_it_still_exists():
+            try:
+                # bash command kills the container if it exists, otherwise return error code 1 without printing an error
+                kill_command = f"docker ps -q --filter \"id={container_id}\" | grep -q . && docker kill {container_id}"
+                removed_container = subprocess.check_output(['bash', '-c', kill_command]).decode("utf-8").strip()
+                # remove_service = subprocess.Popen(["/bin/bash", "-c", "docker-compose down"])
+                print(f"Cleaned up container {removed_container}")
+            except subprocess.CalledProcessError:
+                pass
+
+        atexit.register(kill_container_if_it_still_exists)
+
+    def launch_node(self):
+        """Launches the gazebo world in the docker.
+        Localhost: sending launch command to http server inside container.
+        Container: Launches a ROS node containing a gazebo world, spawns a robot in the world.
+        """
+        try:
+            headers = {'Content-type': 'application/json'}
+            json_data = '{"command_name": "launch_node", "package_name": "gym_offworld_monolith", "launch_file_name":"env_bringup.launch"}'
+            json_data = ast.literal_eval(json_data)
+            result = requests.post("http://127.0.0.1:8008/", data=json.dumps(json_data), headers=headers)
```

Comment:
Please use the container ip address

Reply:
this could be waited until Dylan's branch being merged